src/cleaning_functions.py

```diff
@@ -24,26 +24,29 @@ def get_label_encoded_feature(feature):
     return [label_encoded_feature]
 
 
-def get_one_hot_encoded_feature(feature):
+def get_one_hot_encoded_feature(feature, feature_name):
```

Comment:
I think you got it wrong. Its not iterating through everything and finds the feature that is specified in string and encode that, no.

The encoding function(s) take 2 argument: 
1- A list to encode ( as it was before, nothing changed here )
2- What is the name of the thing that it is encoding

Returns 2 things:
1- Encoded items in M columns.
^ Say, we are talking about binary encoding, the feature is needs at least 7 columns to be properly encoded so it returns MxN list(matrix), M = 7
2- A list of strings that has M elements that you(we) can relate to the name.
^ Say, we are encoding company, so it just returns:
['company_1', 'company_2','company_3'...]

In the end when we encode everything and anything, we initially didnt know which column is what ( since everything is 1 and 0 ). Now we also have another list that is just made of strings in the order of encoded columns. Now you know the 7th column is related to 'company'

The reason why it takes two arguments ( the name of feature ) is, there is no way the function can know what it is that it's encoding. Its just list of strings to be converted to 1s and 0s. 

From your example :

companies = get_encoded(raw_companies)
^ Here companies is the encoded version of raw_companies. Its 7 lists of 1s and 0s, how do you know which columns is where when you concatenate everything ?